### ISSUE TITLE
Remove "Featured" addons support

### ISSUE THREAD

**Issue by eviljeff** [mozilla/addons-server#12907]
The Recommend add-ons programme has supplanted Featured add-ons, and has been live for a number of months (i.e. we're not going back).  It's time to remove the supporting code, tools, and API properties (in a v3 compatible way).  This can be done in stages.

**Comment by AlexandraMoga** [comment]
I see that this patch affected collections, namely the Featured add-ons admin tool. 

Admin tools on -dev no longer have the Featured add-ons option:

**-dev:**
![image](https://user-images.githubusercontent.com/31961530/69867068-73861d00-12ae-11ea-891a-8af3cfc9d60e.png)


**stage:**
![image](https://user-images.githubusercontent.com/31961530/69867080-7a149480-12ae-11ea-9450-f1ac167ed4b9.png)

@eviljeff could this patch have an impact on the featured personas collection? I'm asking because I've noticed that #11487 is reproducible again on -dev. Stage is working fine.
To sum up, https://addons-dev.allizom.org/en-US/firefox/search/?recommended=true&type=statictheme doesn't include all the recommended themes included in https://addons-internal.dev.mozaws.net/en-US/admin/models/bandwagon/collection/2143965/change/

**Comment by eviljeff** [comment]
it shouldn't - being a recommended theme just relies on being in that specific collection.  That the collection isn't a special featured collection any more won't affect that.  ES reindexing does appear broken on -dev though - that's a more likely culprit.

**Comment by AlexandraMoga** [comment]
I understand. I'm going to wait then for the re-indexing to happen and I'll re-test this scenario.

**Comment by AlexandraMoga** [comment]
The issue mentioned above is now also reproducing on stage.
